diff --git a/dlls/winemac.drv/macdrv_main.c b/dlls/winemac.drv/macdrv_main.c
index d89b6d7..0396418 100644
--- a/dlls/winemac.drv/macdrv_main.c
+++ b/dlls/winemac.drv/macdrv_main.c
@@ -708,7 +708,7 @@ static struct dxmt_macdrv_win_data *my_get_win_data(HWND hwnd)
      * It does:
      * get_win_data() -> create_metal_device() -> create_metal_view() -> get_metal_layer() -> release_win_data()
      */
-    client_surface = macdrv_CreateClientSurface(hwnd, 0, FALSE);
+    client_surface = impl_from_client_surface(macdrv_CreateClientSurface(hwnd, 0, FALSE));
 
     /* get_win_data() needs to happen after client_surface creation to avoid deadlocks */
     data = get_win_data(hwnd);
